python/lsst/pipe/tasks/imageDifference.py

```diff
@@ -348,6 +401,39 @@ def makeIdFactory(expId, expBits):
         """
         return afwTable.IdFactory.makeSource(expId, 64 - expBits)
 
+    def runQuantum(self, butlerQC: pipeBase.ButlerQuantumContext,
+                   inputRefs: pipeBase.InputQuantizedConnection,
+                   outputRefs: pipeBase.OutputQuantizedConnection):
+        """Pipelinetask gen3 entry point.
+
+        Parameters
+        ----------
+        expId : `int`
+            Exposure id.
+
+        expBits: `int`
+            Number of used bits in ``expId``.
+
+        Returns
+        -------
+        None
+        """
+        if self.getTemplate.config.coaddName == 'dcr':
+            # TODO DM-22952
+            raise NotImplementedError("TODO DM-22952: Dcr coadd templates are not yet supported in gen3.")
+
+        inputs = butlerQC.get(inputRefs)
+        expId, expBits = butlerQC.quantum.dataId.pack("visit_detector",
+                                                      returnMaxBits=True)
+        inputs['idFactory'] = self.makeIdFactory(expId=expId, expBits=expBits)
+        inputs['templateExposure'] = self.getTemplate.assembleTemplateExposure(
+            butlerQC, inputRefs.skyMap, inputRefs.coaddExposures, inputs['exposure']
+        )
+        del inputs['coaddExposures']
+        del inputs['skyMap']
+        outputs = self.run(**inputs)
```

Comment:
I would suggest defining new variables `idFactory` and `templateExposure` above, and passing the arguments to `self.run()` explicitly, instead manipulating `inputs` so that it unpacks correctly.

Reply:
Sure. Evolved around the `self.run(**inputs)` tutorial example.